packages/framework/aqueduct/src/data-objects/treeDataObject.ts

```diff
@@ -54,12 +54,12 @@ export abstract class TreeDataObject<TTreeView> extends PureDataObject {
 	 * Implementation of SharedTree which is used to generate the view.
 	 * @remarks Created once during initialization.
 	 */
-	#sharedTree: ITree | undefined;
+	#sharedTree: (ITree & ISharedObject) | undefined;
 
 	/**
 	 * Gets the underlying {@link @fluidframework/tree#ITree | tree}.
 	 */
-	public get sharedTree(): ITree {
+	public get sharedTree(): ITree & ISharedObject {
```

Comment:
this seems incorrect. the docs on the type say:

 * Base interface for shared objects from which other interfaces extend.
 * @remarks
 * This interface is not intended to be implemented outside this repository:
 * implementers should migrate to using an existing implementation instead.
 * @privateRemarks
 * Implemented by {@link SharedObjectCore}.

Reply:
seems fine internally, but i don't think the public getter should expose it

Reply:
Hey @anthony-murphy, after going through the doc more thoroughly, I agree with your point. Will close this PR. Thanks!

Reply:
i think the change was fine besides the public exposure on the getter. using it internally made the bind cast better which i thought was fine.